File: Interviews/Striver-SDE-Sheet/Day-17-Binary-Tree/Pre-order-Traversal/144.py
class Solution:
    def preorder(self, root, ans):
        if not root: return

        ans.append(root.val)
        self.preorder(root.left, ans)
        self.preorder(root.right, ans)

    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        self.preorder(root, ans)
        return ans

File: Interviews/LeetCode/Daily-Problems/2023/January/Day-09/144.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
	int val;
	TreeNode* left;
	TreeNode* right;
	TreeNode()
		: val(0), left(nullptr), right(nullptr) {}
	TreeNode(int x)
		: val(x), left(nullptr), right(nullptr) {}
	TreeNode(int x, TreeNode* left, TreeNode* right)
		: val(x), left(left), right(right) {}
};

class Solution {
public:
	void preorder (TreeNode *root, vector<int> &a) {
		if (!root) return;

		a.push_back(root->val);
		preorder (root->left, a);
		preorder (root->right, a);
	}
	vector<int> preorderTraversal(TreeNode* root) {
		vector<int> answer;
		preorder (root, answer);
		return answer;
	}
};